src/ray/raylet/lineage_cache.cc

```diff
@@ -365,31 +365,57 @@ bool LineageCache::UnsubscribeTask(const UniqueID &task_id) {
   return subscribed;
 }
 
-void LineageCache::EvictRemoteLineage(const UniqueID &task_id) {
-  // Remove the ancestor task.
+boost::optional<LineageEntry> LineageCache::EvictTask(const TaskID &task_id) {
   auto entry = lineage_.PopEntry(task_id);
+  if (!entry) {
+    return entry;
+  }
+
+  // Stop listening for notifications about this task.
+  UnsubscribeTask(task_id);
+
+  // Try to flush the children of the committed task. These are the tasks that
+  // have a dependency on the committed task.
+  auto children_entry = uncommitted_ready_children_.find(task_id);
+  if (children_entry != uncommitted_ready_children_.end()) {
+    // Get the children of the committed task that are uncommitted but ready.
+    auto children = std::move(children_entry->second);
+    uncommitted_ready_children_.erase(children_entry);
+
+    // Try to flush the children.  If all of the child's parents are committed,
+    // then the child will be flushed here.
+    for (const auto &child_id : children) {
+      bool flushed = FlushTask(child_id);
```

Comment:
If `FlushTask` returns true, that means that we've begun the process of committing `child_id`, but it hasn't been successfully committed yet. So isn't it premature to remove `child_id` from `uncommitted_ready_tasks_`, since `uncommitted_ready_tasks_` is a mapping from all "tasks that haven't been committed yet", and `child_id` still isn't committed?

If not, then maybe the documentation for `uncommitted_ready_tasks_` and `FlushTask` should be clarified.

Reply:
I'd expect the erase to happen in `HandleEntryCommitted`

Reply:
Ok I think I confused it with `uncommitted_ready_children_`